File: Two Pointers/4. Container With Most Water/solution.scala
object Solution {
    def maxArea(height: Array[Int]): Int = {
        var left = 0
        var right = height.length - 1
        var result = 0
        while (left < right) {
            val newResult = (right - left) * (height(left).min(height(right)))
            result = newResult.max(result)
            if (height(left) > height(right)) {
                right -= 1
            } else {
                left += 1
            }
        }
        result
    }
}

File: Two Pointers/4. Container With Most Water/solution.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut left = 0 as usize;
        let mut right = (height.len() - 1) as usize;
        let mut result = 0;

        while left < right {
            let new_result = (right - left) as i32 * (height[left].min(height[right]));
            result = result.max(new_result);
            if height[left] > height[right] {
                right -= 1;
            } else {
                left += 1;
            }
        }

        result as i32
    }
}
